lib/app/resources/2016_02_02_dogma.md

```diff
@@ -0,0 +1,167 @@
+
+class Attachment {
+  @Serialize.field('id')
+  int id;
+  @Serialize.field('from_url')
+  String fromUrl;
+  @Serialize.field('image_url')
+  String imageUrl;
+}
+``` 
+
+Once you have the following class, you have to export it in a `models.dart` library. `dogma-convert` provides an executable that you can use from a script to generate the converters. See [this](https://github.com/Pacane/SlackHistoryKeeper/blob/master/shared/build.dart) for a sample build script.
+
+The generated encoding/decoding classes look like this:
+```dart
+library slack_history_keeper_shared.src.convert.attachment_convert;
+
+
+//---------------------------------------------------------------------
+// Standard libraries
+//---------------------------------------------------------------------
+import 'dart:convert';
+
+//---------------------------------------------------------------------
+// Imports
+//---------------------------------------------------------------------
+import 'package:dogma_convert/convert.dart';
+import 'package:slack_history_keeper_shared/models.dart';
+
+//---------------------------------------------------------------------
+// Library contents
+//---------------------------------------------------------------------
+
+/// A [ModelDecoder] for [Attachment].
+class AttachmentDecoder extends Converter<Map, Attachment>
+    implements ModelDecoder<Attachment> {
+  @override
+  Attachment create() => new Attachment();
+  @override
+  Attachment convert(Map input, [Attachment model]) {
+    model ??= create();
+
+    model.id = input['id'];
+    model.fromUrl = input['from_url'];
+    model.imageUrl = input['image_url'];
+    return model;
+  }
+}
+
+/// A [ModelEncoder] for [Attachment].
+class AttachmentEncoder extends Converter<Attachment, Map>
+    implements ModelEncoder<Attachment> {
+  @override
+  Map convert(Attachment input) {
+    var model = {};
+
+    model['id'] = input.id;
+    model['from_url'] = input.fromUrl;
+    model['image_url'] = input.imageUrl;
+
+    return model;
+  }
+}
+```
+
+You may (or may not if you don't like the convention) add a `toJson` method that delegates to new generated Encoder to your original model class to integrate easily with existing packages that use JSON Codec (like Redstone for example).
+
+On client side apps, you can use the Decoders like this:
+
+```dart
+final UserDecoder userDecoder = new UserDecoder();
+
+Future<List<User>> fetchUsers() async {
+  var result = await client.get('$apiUrl/users');
+
+  var json = JSON.decode(result.body) as List<Map>;
+
+  var users = json.map((Map m) => userDecoder.convert(m)).toList();
+
+  return users;
+}
+```
+
+## Conclusion
+So if you want keep a nice small code size for your web app, or even to have a clean separation of concern, with low maintenance for JSON serialization, I'd recommend to use Dogma.
+
+If has been pretty easy for me to generate the mapping code between Dart and JSON, and use the converters in my code base.
```

Comment:
separation of CONCERNS

I'd recommend YOU to use Dogma.

Little bit of a run on sentence actually.